### PR TITLE
feat: Add support for EXIT PERFORM statement for CCF

### PR DESCRIPTION
**Pull request by Nurkambay**
## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

https://www.ibm.com/docs/en/cobol-zos/6.4?topic=statement-format-5-inline-perform

### CODE REVIEW

**Comment by slavek-kucera** [comment]
Do you want to detect the following in CCF?
```
If you specify an EXIT PERFORM statement outside of an inline PERFORM statement, the EXIT PERFORM is ignored.
```

**Comment by Nurkambay** [comment]
> Do you want to detect the following in CCF?
> 
> ```
> If you specify an EXIT PERFORM statement outside of an inline PERFORM statement, the EXIT PERFORM is ignored.
> ```

Yep, it will be detected in CCF. COBOL LS should just provide CFAST Tree for CCF and CCF will do the rest. I don't want to spread logic to filter statements on the COBOL LS side, it is not COBOL LS responsibility to decide what to do with a statement

File: server/engine/src/main/resources/resourceBundles/messages_en.properties [hunk 1]
```diff
@@ -75,6 +75,7 @@ semantics.globalNon01Level=GLOBAL can only be used on level 01
 semantics.globalTooManyDefinitions=Global variable must have a unique name
 semantics.notDefined=Variable %s is not defined
 semantics.ambiguous=Ambiguous reference for %s
+semantic.exitPerformIsIgnored=The EXIT PERFORM statement is outside of an inline PERFORM statement and will be ignored
```
Comment (Nurkambay) [inline server/engine/src/main/resources/resourceBundles/messages_en.properties:78]: @zeibura please check this sentence

Comment (zeibura) [reply server/engine/src/main/resources/resourceBundles/messages_en.properties:78]: Fine

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/ExitPerformEnricher.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * Adds an information is EXIT PERFORM statement situated inside or outside inline perform block
+ */
+public class ExitPerformEnricher implements Processor<ExitPerformNode> {
+  @Override
+  public void accept(ExitPerformNode exitPerformNode, ProcessingContext processingContext) {
+      Node parent = exitPerformNode.getParent();
+      while (parent != null) {
+          if (parent instanceof PerformNode) {
```
Comment (slavek-kucera) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/ExitPerformEnricher.java:32]: This should probably stop at the ProgramNode, right?

Comment (Nurkambay) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/ExitPerformEnricher.java:32]: I am not sure.

Comment (Nurkambay) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/ExitPerformEnricher.java:32]: But let's assume it is correct

Comment (slavek-kucera) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/ExitPerformEnricher.java:32]: Nevermind... program cannot be nested in the middle of the statement.

**Review by slavek-kucera — APPROVED** [review]

**Pull request merged**